vmdk_plugin/utils/fs/fs_windows.go

```diff
@@ -0,0 +1,242 @@
+		Get-PnpDevice -FriendlyName 'VMware Virtual disk SCSI Disk Device' |
+		Select-Object -ExpandProperty InstanceId |
+		ForEach-Object {
+			$instanceId = $_
+			$uiNum = (
+				Get-PnpDeviceProperty -InstanceId $instanceId -KeyName 'DEVPKEY_Device_UINumber' |
+				Select -expand Data -erroraction 'silentlycontinue'
+			)
+			$addr = (
+				Get-PnpDeviceProperty -InstanceId $instanceId -KeyName 'DEVPKEY_Device_Address' |
+				Select -expand Data -erroraction 'silentlycontinue'
+			)
+			If ($uiNum -eq '%s' -And $addr -eq '%s') {
+				Write-Host ""
+				Get-WmiObject Win32_DiskDrive |
+				Where-Object { $_.PNPDeviceId -eq $instanceId } |
+				Select-Object -ExpandProperty Index
+				exit
+			}
+		}
+		Write-Host ""
+		Write-Host "DiskNotFound"
+	`
+
+	// DiskPart script to format a disk with a filesystem.
+	formatDiskScript = `
+		select disk %s
+		clean
+		online disk
+		attribute disk clear readonly
+		create partition primary
+		select partition 1
+		active
+		format fs=%s label="%s" quick
+		exit
+	`
+)
+
+// VerifyFSSupport checks whether the fstype filesystem is supported.
+func VerifyFSSupport(fstype string) error {
+	if fstype != ntfs {
+		log.WithFields(log.Fields{"fstype": fstype}).Error("Unsupported fstype ")
+		return fmt.Errorf("Not found mkfs for %s\nSupported filesystems: %s",
+			fstype, ntfs)
+	}
+	return nil
+}
+
+// DevAttachWaitPrep initializes and returns a new DiskWatcher.
+func DevAttachWaitPrep() (*DeviceWatcher, error) {
+	watcher := NewDeviceWatcher()
+	watcher.Init()
+	return watcher, nil
+}
+
+// DevAttachWait waits until the specified disk is attached, or returns
+// an error on watcher failure.
+func DevAttachWait(watcher *DeviceWatcher, volDev *VolumeDevSpec) error {
+	defer watcher.Terminate()
+	for {
+		log.WithFields(log.Fields{"volDev": *volDev}).Info("Waiting for a watcher event ")
+		select {
+		case event := <-watcher.Event:
+			log.WithFields(log.Fields{"volDev": *volDev,
+				"event": event}).Info("Watcher emitted an event ")
+			diskNum, err := getDiskNum(volDev)
+			if err != nil {
+				log.WithFields(log.Fields{"volDev": *volDev,
+					"err": err}).Warn("Couldn't map volDev to diskNum, continuing.. ")
+			} else {
+				log.WithFields(log.Fields{"volDev": *volDev,
+					"diskNum": diskNum}).Info("Successfully mapped volDev to diskNum ")
+				return nil
+			}
+			log.WithFields(log.Fields{"volDev": *volDev}).Warn("Couldn't locate disk, waiting.. ")
+
+		case err := <-watcher.Error:
+			log.WithFields(log.Fields{"volDev": *volDev,
+				"err": err}).Error("Watcher returned an error ")
+			return err
+
+		case <-time.After(maxDiskAttachWaitSec):
+			msg := "Disk mapping timed out "
+			log.WithFields(log.Fields{"volDev": *volDev}).Error(msg)
+			return errors.New(msg)
+		}
+	}
+}
+
+// DevAttachWaitFallback is a NOP.
+func DevAttachWaitFallback() {
+	// NOP since DevAttachWaitPrep never returns an error
+}
+
+// Mkdir creates a directory at the specified path.
+func Mkdir(path string) error {
+	return nil
```

Comment:
This isn't yet implemented. We'd be implementing this while working on volume attach.